command/agent/config/config.go

```diff
@@ -43,7 +43,8 @@ type Vault struct {
 
 // Cache contains any configuration needed for Cache mode
 type Cache struct {
-	UseAutoAuthToken bool `hcl:"use_auto_auth_token"`
+	UseAutoAuthToken        bool `hcl:"use_auto_auth_token"`
+	UseAutoAuthTokenEnforce bool `hcl:"use_auto_auth_token_enforce"`
```

Comment:
Feels like a better UX would be to use an interface and then at runtime figure out if it's a bool or string, so that it can be true, false, "true", "false", "force".

Reply:
Can you elaborate your vision of a tertiary value set?  I'm a bit unclear as to how it could be anything other than a boolean "yes" or "no".  We either instruct the agent to always use it's token regardless of what the client sends, or we proxy whatever we get from the client.  Am I missing something?

Reply:
I *believe* the suggestion is to change this to be like `TLSSkipVerify` and `TLSSkipVerifyRaw` from the `Vault` type:

https://github.com/hashicorp/vault/blob/f6185270ab15cf007a1b9bfe85eb4e3782dce331/command/agent/config/config.go#L37-L38

That then gets filtered through with `parseutil.ParseBool()`:

https://github.com/hashicorp/vault/blob/f6185270ab15cf007a1b9bfe85eb4e3782dce331/command/agent/config/config.go#L190-L195

Reply:
Oh, you mean "re-purpose" the `UseAutoAuthToken` field instead of adding another one?

Reply:
Hey - I chatted with @jefferai and we thought that instead of a new option, instead it may be a better UX to change `use_auto_auth_token` to accept `true`, `false`, and a new option `force`. The first two values would behave as it does today, where a token in the request would override the auto-auth value (if used), and the new option `force` would behave as you're looking for, where it forces the auto-auth value and ignores any token in the request. 

Following the `TLSSkipVerify` and `TLSSkipVerifyRaw` pattern would allow us to parse and match the value to one of those three options. This way users only have 1 configuration option and don't have to manage the combination of the two to get the desired effect.

Thoughts on this approach?